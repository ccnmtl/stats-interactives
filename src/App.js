import { CentralLimitGraph } from './CentralLimitGraph'
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

export const App = () => (
    <Router>
        <div>
            <h1>Graphs</h1>
            <div className="container">
                <nav>
                    <ul>
                        <li>
                            <Link to="/">Home</Link>
                        </li>
                        <li>
                            <Link to="/central-limit-theorem">Central Limit Theorem</Link>
                        </li>
                    </ul>
                </nav>
            </div>
            <div className="container">
                <Route exact path="/" component={Home} />
                <Route path="/central-limit-theorem" component={CentralLimitGraph} />
            </div>
        </div>
    </Router>
)

const Home = () => (
    <h1>Home!</h1>
)
