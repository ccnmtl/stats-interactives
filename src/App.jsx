/*eslint max-len: ["error", { "ignoreStrings": true }]*/
import React from 'react';
import PropTypes from 'prop-types';
import { CentralLimitGraph } from './CentralLimitGraph';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

export const App = () => (
    <Router>
        <main role="main">
            <Route exact path="/" component={Home} />
            <Route path="/central-limit-theorem"
                component={CentralLimitGraph} />
        </main>
    </Router>
);

const PreviewData = [
    {
        'slug': '/central-limit-theorem',
        'title': 'Central Limit Theorem',
        'text': 'Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui.'
    },
    {
        'slug': '/ols-regression',
        'title': 'OLS Regression',
        'text': 'Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui.'
    },
];

const PreviewBlock = (props) => (
    <div key={props.slug} className="col-md-4">
        <h2>
            <Link to={props.slug}>{ props.title }</Link>
        </h2>
        <p>{props.text}</p>
    </div>
);

const Preview = () => (
    PreviewData.map(PreviewBlock)
);

const Home = () => (
    <>
    <div className="jumbotron">
    
        <div className="container">
        
            <h1 className="display-3">
                Theorem7
            </h1>
            
            <p className="lead">
                Seven Interactive Simulations of Statistics Theorems for Deeper Understanding
            </p>

            <p>
                Specifically designed for Doru Cojoc&#39;s Quantitative
                Analysis I course at SIPA, these simulations are intended as a
                test of effectiveness at improving learning outcomes. This
                project seeks to address this by turning them into web
                interactives, openly available. These interactives will be
                provided to students for their own study and experimentation,
                which can be used here or embedded/linked elsewhere.
            </p>
            
            <p>
                <a className="btn btn-primary btn-lg" href="https://sipa.columbia.edu/faculty-research/faculty-directory/doru-c-cojoc" role="button">Contact Dr. Cojoc</a>
            </p>
            
        </div>
        
    </div>
    
    <div className="container">
    
        <div className="row">
        
            <Preview />
            
        </div>
        
    </div>
    </>
);

PreviewBlock.propTypes = {
    slug: PropTypes.string,
    title: PropTypes.string,
    text: PropTypes.string
};
export {PreviewBlock, Preview};
