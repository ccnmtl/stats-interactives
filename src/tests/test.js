import React, {Component} from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Nav, Preview} from '../App.jsx';

test('Nav elements render as expected', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Nav />
        </MemoryRouter>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('Preview elements render as expected', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Preview />
        </MemoryRouter>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
