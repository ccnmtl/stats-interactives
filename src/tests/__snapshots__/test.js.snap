// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav elements render as expected 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-darkt"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Theoremicious
  </a>
  <div
    className="navbar-nav-scroll"
  >
    <ul
      className="navbar-nav bd-navbar-nav flex-row"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/central-limit-theorem"
          onClick={[Function]}
        >
          Central Limit Theorem
        </a>
      </li>
    </ul>
  </div>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
     
    <span
      className="navbar-toggler-icon"
    />
     
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  />
</nav>
`;

exports[`Preview elements render as expected 1`] = `
Array [
  <div
    className="col-md-4"
  >
    <h2>
      <a
        href="/central-limit-theorem"
        onClick={[Function]}
      >
        Central Limit Theorem
      </a>
    </h2>
    <p>
      Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui.
    </p>
  </div>,
  <div
    className="col-md-4"
  >
    <h2>
      <a
        href="/ols-regression"
        onClick={[Function]}
      >
        OLS Regression
      </a>
    </h2>
    <p>
      Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui.
    </p>
  </div>,
]
`;
